### ISSUE TITLE
Return from forEach is not behaving as expected

### ISSUE THREAD

**Issue by samreid** [phetsims/isotopes-and-atomic-mass#104]
Similar to the problem in https://github.com/phetsims/gene-expression-essentials/issues/138, discovered in https://github.com/phetsims/chipper/issues/1033.  It looks like this code is supposed to short circuit when a match is found, but it iterates over everything anyways.

```js
  checkForParticleOverlap() {
    let overlapCheck = false;

    this.containedIsotopes.forEach( isotope1 => {
      for ( let i = 0; i < this.containedIsotopes.length; i++ ) {
        const isotope2 = this.containedIsotopes.get( i );
        if ( isotope1 === isotope2 ) {
          // Same isotope so skip it!
          continue;
        }

        const distance = isotope1.positionProperty.get().distance( isotope2.positionProperty.get() );
        if ( distance < isotope1.radiusProperty.get() + isotope2.radiusProperty.get() ) {
          overlapCheck = true;
          return overlapCheck;
        }
      }
    } );

    return overlapCheck;
  }
```

**Comment by samreid** [comment]
Similar problem in this code:

```js
  getNumericalControllerForIsotope( isotope ) {
    let isotopeController = null;
    this.numericalControllerList.forEach( controller => {
      if ( controller.isotopeConfig.equals( isotope ) ) {
        // Found it.
        isotopeController = controller;
        return isotopeController;
      }
    } );
    return isotopeController;
  }
```

**Comment by jbphet** [comment]
I fixed the first bit of code by using standard C-style loops and using the overlap flag to drop out of them.  I tested this via explicit overlap testing and fuzz testing.

The second bit of code doesn't seem to be used, so I simply deleted this.

I don't think a follow-up review is necessary so I'm just going to close this.  @samreid - If you see this go by, you're welcome to review my changes and reopen if you think it's justified.
